test/tests/test-mem-layout.cc

```diff
@@ -0,0 +1,75 @@
+#include <iostream>
+#include <cstdlib>
+#include <ctime>
+#include <limits>
+#include <cmath>
+
+#include <UnitTesting.h>
+#include "marlin/book/MemLayout.h"
+#include "ROOT/RHist.hxx"
+#include "ROOT/RAxis.hxx"
+
+using namespace ROOT::Experimental;
+using namespace marlin::book;
+struct Type1 {
+	std::array<int, 5> bins;
+	Type1(int a0, int a1, int a2, int a3, int a4)
+		: bins{a0, a1, a2, a3, a4} {}
+};
+
+void MergeType1(const std::shared_ptr<Type1>& p1, const std::shared_ptr<Type1>& p2) {
+	for(int i = 0; i < 5; ++i) {
+		p1->bins[i] += p2->bins[i];
+	}
+}
+template<typename ... Args_t>
+class SharedType1 : public SharedMemLayout<Type1, MergeType1, Args_t ... > {
```

Comment:
Why to you need to sub-class here ?

Reply:
for binding the merge function with the class (for Type1), we don't really need it.